tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/JmapSettingsSetMethod.scala

```diff
@@ -97,7 +104,25 @@ class SettingsSetRequestSetMethod @Inject()(val jmapSettingsRepository: JmapSett
       oldState <- evaluateOldState(updateResults, currentStateIfAllUpdatesFailed)
       newState <- evaluateNewState(updateResults, currentStateIfAllUpdatesFailed)
       response = createResponse(invocation.invocation, request, updateResults, oldState, newState)
-    } yield InvocationWithContext(response, invocation.processingContext)
+    } yield {
+      dispatchSettingsChangeEvent(mailboxSession.getUser, oldState, newState).subscribe()
+
+      InvocationWithContext(response, invocation.processingContext)
```

Comment:
Why not just flatMap?

I do not see why we need to be async here.

Reply:
I thought about 2 points:
- sync makes the response slower (even dispatch event should be quick, but still slower)
- In the case RabbitMQ is down, then failure to dispatch an event to RabbitMQ forces the Settings/set to fail. Seems not that critical to me.

Reply:
> sync makes the response slower (even dispatch event should be quick, but still slower)

IMO not a concern we should tackle here but more at the event bus layer.

Async here means anyway unbounded memory queue. Not sure that's a great idea...

> In the case RabbitMQ is down, then failure to dispatch an event to RabbitMQ forces the Settings/set to fail. Seems not that critical to me.

Again an event bus concern.

Event bus is supposed to save failure to dispatch into deadletter so I think that this concern is even already handled.